sql/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -228,6 +230,16 @@ message Expression {
       DataType struct_type = 1;
       repeated Literal elements = 2;
     }
+
+    message SpecializedArray {
+      DataType element_type = 1;
```

Comment:
Since the values can only be of one type should this rather be:

```
oneof value_type {
  repeated bool booleans = 2;
  ...
}
```

Reply:
I mean unfortunately, this would rather look like

```
message SpecializedArray {
   
  oneof value_type {
    Bools bools = 1;
    Ints ints = 2;
    // ...
  }

  message Bools {
    repeated bool values = 1;
  }

  message Ints {
    repeated int32 values = 1;
  }
}
```

Reply:
Which I would personally prefer because it makes clear that there can only be one kind of type in the array.

Reply:
Sounds good